numpy/core/src/multiarray/compiled_base.c

```diff
@@ -803,17 +803,25 @@ arr_interp_complex(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwdict)
     return NULL;
 }
 
+
+#define EMPTY_SEQUENCE_ERR_MSG "indices must be integral: the provided " \
+    "empty sequence was inferred as float. Wrap it with " \
+    "'np.array(indices, dtype=np.intp)'"
+
+#define NON_INTEGRAL_ERROR_MSG "only int indices permitted"
 /*
  * Converts a Python sequence into 'count' PyArrayObjects
  *
- * seq       - Input Python object, usually a tuple but any sequence works.
- * op        - Where the arrays are placed.
- * count     - How many arrays there should be (errors if it doesn't match).
- * paramname - The name of the parameter that produced 'seq'.
+ * seq         - Input Python object, usually a tuple but any sequence works.
+ * paramname   - The name of the parameter that produced 'seq'.
+ * count       - How many arrays there should be (errors if it doesn't match).
+ * op          - Where the arrays are placed.
  */
 static int sequence_to_arrays(PyObject *seq,
```

Comment:
This function name and its docs should be changed to indicate that it handles arrays of indices / that they must be integers / something like that

Reply:
changed